packages/java-edition/src/mcfunction/checker/index.ts

```diff
@@ -81,43 +88,69 @@ const entity: core.SyncChecker<EntityNode> = (node, ctx) => {
 	}
 }
 
-const itemStack: core.SyncChecker<ItemStackNode> = (node, ctx) => {
+const itemPredicate: core.SyncChecker<ItemPredicateNode> = (node, ctx) => {
 	if (node.nbt) {
 		nbt.checker.index(
 			'minecraft:item',
 			core.ResourceLocationNode.toString(node.id, 'full'),
+			{ isPredicate: true },
 		)(node.nbt, ctx)
 	}
-	if (node.components) {
-		const groupedComponents = new Map<
-			string,
-			core.PairNode<core.ResourceLocationNode, nbt.NbtNode>[]
-		>()
-
-		node.components!.children.forEach(component => {
-			const componentName = core.ResourceLocationNode.toString(
-				component.key!,
-				'full',
-			)
-
-			if (!groupedComponents.has(componentName)) {
-				groupedComponents.set(componentName, [])
+	if (!node.tests) {
+		return
+	}
+	for (const anyOfTest of node.tests.children) {
```

Comment:
i don't understand this `tests` field. can you explain it? (i'm also probably not terribly familiar with item components tbh)

Reply:
`tests` is a field of `ItemPredicateNode`.